framework/src/auxkernels/VectorMagnitudeAux.C

```diff
@@ -32,6 +32,15 @@ VectorMagnitudeAux::VectorMagnitudeAux(const InputParameters & parameters)
     _y(_mesh.dimension() >= 2 ? coupledValue("y") : _zero),
     _z(_mesh.dimension() >= 3 ? coupledValue("z") : _zero)
 {
+  auto require_qp_calcs = [this](const auto & var_name)
+  {
+    if (isCoupled(var_name))
+      getFieldVar(var_name, 0)->requireQpComputations();
```

Comment:
any change we can move all this to the base class for auxkernels?
we can check which functors are requested, and request the QpComputations

Reply:
This isn't a functor though in this derived class. And many aux kernels now have functor versions which do not need qp computations. You could be registering false positives for needing qp computations and incurring computational cost that is not needed if you just made all coupled variables in every aux kernel `requireQpComputations`

Reply:
yeah I suppose I worded this wrong. 

We can check which finite volume variables / other objects needing explicit Qp computations are being queried through a non-functor interface. Then we can requireQpComputations in the base class ?

Reply:
Alright I've updated `Coupleable` such that if a finite volume variable is requested through `add.*CoupledVar` then we `requireQpComputations`. So someone wishing to not trigger pre-init'd qp calculations should use `addParam<MooseFunctorName>`